src/mono/mono/mini/aot-compiler.c

```diff
@@ -4208,22 +4208,24 @@ get_plt_entry (MonoAotCompile *acfg, MonoJumpInfo *patch_info)
 			res->debug_sym = get_plt_entry_debug_sym (acfg, res->ji, acfg->plt_entry_debug_sym_cache);
 		if (synchronized) {
 			/* Avoid duplicate symbols because we don't cache */
-			res->symbol = g_strdup_printf ("%s_%d", res->symbol, synchronized_symbol_idx);
-			if (res->debug_sym)
-				res->debug_sym = g_strdup_printf ("%s_%d", res->debug_sym, synchronized_symbol_idx);
+			char *tmp_symbol = res->symbol;
+			res->symbol = mono_mempool_strdup_printf (acfg->mempool, "%s_%d", res->symbol, synchronized_symbol_idx);
+			g_free (tmp_symbol);
+			if (res->debug_sym) {
+				char *tmp_debug_sym = res->debug_sym;
+				res->debug_sym = mono_mempool_strdup_printf (acfg->mempool, "%s_%d", res->debug_sym, synchronized_symbol_idx);
+				g_free (tmp_debug_sym);
```

Comment:
`get_plt_entry_debug_sym` calls `sanitize_symbol` which calls `mono_mempool_strdup`, so it's not ok to always call  `g_free (tmp_debug_sym)`

Reply:
Interesting, this is upstreamed from .net8 implementation so maybe that implementation changed, need to investigate and adjust that path.

Reply:
OK, will drop that g_free, since changes done in get_plt_entry_debug_sym makes sure to free any heap allocated memory  if we end up returning a mempool allocated string.

Reply:
I changed `sanitize_symbol `to use `g_strdup` instead of `mempool` since there is no clear ownership of the passed in string and what "type" of allocation that gets returned back to caller. Since the majority of the current implementation already used none mempool allocated strings it made sense to just move over to use `g_strdup` instead of `mempool` and keep the free as implemented in above code.